### DISCUSSION TITLE
Map list of ids to list of objects with id property

### DISCUSSION THREAD

**Discussion by kolpav** [LuckyPennySoftware/AutoMapper#3769] [NA]
I have run out of places where to ask sorry for spamming issues with questions.

### Source/destination types

Source: `Guid`
Destination: objects implementing `interface IEntity { Guid Id { get; set; } }`

### Mapping configuration

I have tried a lot of solutions mainly configuring `ForAllPropertyMaps` where I am checking if source is `List<Guid>` and destination implements `IEntity` but I can't seem to get it right.

### Versions
```
<PackageReference Include="AutoMapper" Version="10.1.1" />
<PackageReference Include="AutoMapper.Extensions.Microsoft.DependencyInjection" Version="8.1.1" />
```

### Expected behavior

```cs
interface IEntity
{
  Guid Id { get; set; }
}

public class User : IEntity
{
  public Guid Id { get; set; }
  public List<Comment> Comments { get; set; }
}

public class CreateUserCommand
{
  public Guid Id { get; set; }
  public List<Guid> CommentIds { get; set; } // Ids of comments which are already in db
}

CreateMap<CreateUserCommand, User>()
  .ForMember(x => x.Comments, opt => opt.MapFrom(x => x.CommentIds));


var user = mapper.Map<User>(createUserCommand);
```

Mapping I am trying to achieve is:
```cs
(CreateUserCommand command) => new User
{
  Id = command.Id,
  Comments = command.CommentIds.Select(id => new Comment { Id = id })
}
```
its shown for concrete type `User` but I would like it to work for any class implementing `IEntity` interface meaning class that has `Guid Id`

### Actual behavior

Mapping throws an error.

### Steps to reproduce

Here is fiddle with one of my many attempts

https://dotnetfiddle.net/Widget/vvmnog

I think I am trying to achieve something which should be very common your DTOs contains list of ids but your entity has list of entities and I would like to map to them such that they have only `Id` property set which is all you need to provide to entity framework db context.

Here is example command handler for type of code I have in my app

```cs
public async Task<User> Handle(CreateUserCommand command, CancellationToken cancellationToken)
{
  var user = _mapper.Map<User>(command);
  _context.Comments.AttachRange(user.Comments);
  await _context.Users.AddAsync(user, cancellationToken);
  await _context.SaveChangesAsync(cancellationToken);
  return user
}
```

I was able to make it work by handwriting map for every class implementing `IEntity` like so:
```cs
.ForMember(x => x.Comments, opt => opt.MapFrom(x => x.CommentIds.Select(id => new Comment { Id = id })));
```

but I feel there must be more generic solution.

**Comment by Cubelaster** [comment]
You can try this, or something along those lines. 
https://docs.automapper.org/en/stable/Custom-type-converters.html

There are more examples on the docs.
